### PR TITLE
remove fips-specific patches/config and code

### PR DESCRIPTION
**Pull request by tobiasb-ms**
###### Merge Checklist  
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
The FIPS-specific things on `openssl.spec` (and related patches) are all RHEL-specific. I addition, our `FIPS` compliance will be handled by `SymCrypt`. So this change removes all of that `FIPS`-specific stuff.

###### Change Log  
- Remove the various patches that implement/augment `FIPS`.
- Remove `FIPS`-specific config and code.

###### Does this affect the toolchain?  
**YES**

###### Test Methodology
- Built locally
- Pipeline: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=484503&view=results

### CODE REVIEW

File: SPECS/openssl/openssl.spec [hunk 1]
```diff
@@ -51,94 +44,18 @@ Patch11:  0011-Remove-EC-curves.patch
 Patch12:  0012-Disable-explicit-ec.patch
 # # Skipped tests from former 0011-Remove-EC-curves.patch
 Patch13:  0013-skipped-tests-EC-curves.patch
-# # Instructions to load legacy provider in openssl.cnf
+# # Instructions to load ly provider in openssl.cnf
```
Comment (christopherco) [inline SPECS/openssl/openssl.spec:47]: nit - looks like an unintended diff

File: SPECS/openssl/openssl.spec [hunk 2]
```diff
@@ -51,94 +44,18 @@ Patch11:  0011-Remove-EC-curves.patch
...
-# AZL: NOTE: Removed this because it is RHEL-specific, but want a record of it until I do the final
-#            review of patches.
-#Patch45:  0045-FIPS-services-minimize.patch
-# # Execute KATS before HMAC verification
-Patch47:  0047-FIPS-early-KATS.patch
 # # Selectively disallow SHA1 signatures rhbz#2070977
 # AZL: NOTE: Had to change this patch because of cascading changes from previous AZL note(s)
 Patch49:  0049-Allow-disabling-of-SHA1-signatures.patch
 # # Support SHA1 in TLS in LEGACY crypto-policy (which is SECLEVEL=1)
 Patch52:  0052-Allow-SHA1-in-seclevel-1-if-rh-allow-sha1-signatures.patch
```
Comment (christopherco) [inline SPECS/openssl/openssl.spec:56]: I am curious if we can remove this patch in a future PR. My thinking:
1. We shouldn't allow SHA1 generally
2. We don't supply a crypto-policy setting like Fedora or RHEL

File: SPECS/openssl/openssl.spec (hunk 1 above)
Comment (christopherco) [inline SPECS/openssl/openssl.spec:47]: We should review this patch to see if we should keep or remove it. The default still uses secure crypto with this patch applied, but removing this patch also removes a user's ability to easily activate the less secure crypto (unless they know the specific setting so this suggestion is closer to obfuscation)

**Review by christopherco — APPROVED** [review]

File: SPECS/openssl/openssl.spec (hunk 2 above)
Comment (tobiasb-ms) [reply SPECS/openssl/openssl.spec:56]: we almost certainly can; I'm targeting FIPS specifically with this change though

**Pull request merged**